### DISCUSSION TITLE
Cleaning cache flag?

### DISCUSSION THREAD

**Discussion by stefanos82** [edubart/nelua-lang#95] [Q&A] [answered]
The following flag `-C, --no-cache        Don't use any cached compilation` makes it clear of what it does; the question is though, how do I clean the already generated C code or compiled binary?

I usually go to `$HOME/.cache/nelua/` and clean my content **or** when I recompile the latest pulled commit, `make clean` does it for me.

Should not there be a `cleanup-cache` flag of some sort?

**Comment by edubart** [comment, marked as answer]
First you don't need to clean the cache after updating the Nelua compiler, because the cache will be invalidated anyway if any pull change impacts the code.

There is no flag to clean the cache folder, and I don't see much need for that at the moment. Also this is not so simple to do, there is no standard cross platform API to recursively remove directories, and is also dangerous to remove files recursively, this problem is out of the scope of the compiler. If you want to clean the cache indeed you have to remove the directory `$HOME/.cache/nelua/` .
